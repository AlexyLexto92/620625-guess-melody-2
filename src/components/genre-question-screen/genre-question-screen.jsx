import React, {PureComponent} from 'react';
import PropTypes from 'prop-types';
import {AudioPlayer} from '../audioPlayer/audioPlayer.jsx';

export class GenreQuestionScreen extends PureComponent {
  constructor(props) {
    super(props);
    this.state = {
      activePlayer: -1,
    };
  }

  render() {
    const {question, onAnsverHendler, checkboxHandler} = this.props;
    const {answers, genre} = question;
    return <section className="game game--genre">
      <header className="game__header">
        <a className="game__back" href="#">
          <span className="visually-hidden">Сыграть ещё раз</span>
          <img className="game__logo" src="img/melody-logo-ginger.png" alt="Угадай мелодию" />
        </a>

        <div className="timer__value">
          <span className="timer__mins">05</span>
          <span className="timer__dots">:</span>
          <span className="timer__secs">00</span>
        </div>

        <div className="game__mistakes">
          <div className="wrong"></div>
          <div className="wrong"></div>
          <div className="wrong"></div>
        </div>
      </header>

      <section className="game__screen">
        <h2 className="game__title">Выберите {genre} треки</h2>
        <form className="game__tracks" onSubmit={(evt) => {
          evt.preventDefault();
          onAnsverHendler();
        }}>
          {answers.map((elem, it) =>
            <div className="track" key={`answer-${it}`}>
              <AudioPlayer
                src={elem.src}
                isPlaying={it === this.state.activePlayer}
                onPlayButtonClick={() => this.setState({
                  activePlayer: this.state.activePlayer === it ? -1 : it
                })}
              />

              <div className="game__answer">
                <input className="game__input visually-hidden"
                  type="checkbox" name={`answer-${it}`} value={elem.genre} id={it} onChange={checkboxHandler} />
                <label className="game__check" htmlFor={it}>Отметить</label>
              </div>
            </div>
          )}
          <button className="game__submit button" type="submit" >Ответить</button>
        </form>
      </section>
    </section >;
  }
}

GenreQuestionScreen.propTypes = {
  onAnsverHendler: PropTypes.func,
  checkboxHandler: PropTypes.func,
  question:
    PropTypes.shape({
      type: PropTypes.string,
      genre: PropTypes.string,
      answers: PropTypes.arrayOf(
          PropTypes.shape({
            src: PropTypes.string,
            genre: PropTypes.string
          })
      )
    })
};
