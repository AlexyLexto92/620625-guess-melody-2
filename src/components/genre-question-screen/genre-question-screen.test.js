import React from 'react';
import renderer from 'react-test-renderer';
import {GenreQuestionScreen} from './genre-question-screen.jsx';
import {createNodeMock} from '../test/tests.js';

it(`App correctly renders after relaunch`, () => {
  const options = {createNodeMock};
  const tree = renderer.create(<GenreQuestionScreen
    count={4}
    time={7}
    onClick={jest.fn()}
    question={
      {
        type: `genre`,
        genre: `rock`,
        answers: [{
          src: `https://upload.wikimedia.org/wikipedia/commons/1/1f/Uganda_flag_and_national_anthem_-_Oh_Uganda_Land_o.ogg`,
          genre: `rock`,
        },
        {
          src: `https://upload.wikimedia.org/wikipedia/commons/1/1f/Uganda_flag_and_national_anthem_-_Oh_Uganda_Land_o.ogg`,
          genre: `pop`,
        },
        {
          src: `https://upload.wikimedia.org/wikipedia/commons/1/1f/Uganda_flag_and_national_anthem_-_Oh_Uganda_Land_o.ogg`,
          genre: `jazz`,
        },
        {
          src: `https://upload.wikimedia.org/wikipedia/commons/1/1f/Uganda_flag_and_national_anthem_-_Oh_Uganda_Land_o.ogg`,
          genre: `rock`,
        }
        ],
      }
    }
  />,
  options
  )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
