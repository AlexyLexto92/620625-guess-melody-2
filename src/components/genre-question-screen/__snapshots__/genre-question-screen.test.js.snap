// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App correctly renders after relaunch 1`] = `
<section
  className="game game--genre"
>
  <header
    className="game__header"
  >
    <a
      className="game__back"
      href="#"
    >
      <span
        className="visually-hidden"
      >
        Сыграть ещё раз
      </span>
      <img
        alt="Угадай мелодию"
        className="game__logo"
        src="img/melody-logo-ginger.png"
      />
    </a>
    <div
      className="timer__value"
    >
      <span
        className="timer__mins"
      >
        05
      </span>
      <span
        className="timer__dots"
      >
        :
      </span>
      <span
        className="timer__secs"
      >
        00
      </span>
    </div>
    <div
      className="game__mistakes"
    >
      <div
        className="wrong"
      />
      <div
        className="wrong"
      />
      <div
        className="wrong"
      />
    </div>
  </header>
  <section
    className="game__screen"
  >
    <h2
      className="game__title"
    >
      Выберите 
      rock
       треки
    </h2>
    <form
      className="game__tracks"
      onSubmit={[Function]}
    >
      <div
        className="track"
      >
        <button
          className="track__button track__button--play"
          disabled={true}
          onClick={[Function]}
          type="button"
        />
        <div
          className="track__status"
        >
          <audio />
        </div>
        <div
          className="game__answer"
        >
          <input
            className="game__input visually-hidden"
            id={0}
            name="answer-0"
            type="checkbox"
            value="rock"
          />
          <label
            className="game__check"
            htmlFor={0}
          >
            Отметить
          </label>
        </div>
      </div>
      <div
        className="track"
      >
        <button
          className="track__button track__button--play"
          disabled={true}
          onClick={[Function]}
          type="button"
        />
        <div
          className="track__status"
        >
          <audio />
        </div>
        <div
          className="game__answer"
        >
          <input
            className="game__input visually-hidden"
            id={1}
            name="answer-1"
            type="checkbox"
            value="pop"
          />
          <label
            className="game__check"
            htmlFor={1}
          >
            Отметить
          </label>
        </div>
      </div>
      <div
        className="track"
      >
        <button
          className="track__button track__button--play"
          disabled={true}
          onClick={[Function]}
          type="button"
        />
        <div
          className="track__status"
        >
          <audio />
        </div>
        <div
          className="game__answer"
        >
          <input
            className="game__input visually-hidden"
            id={2}
            name="answer-2"
            type="checkbox"
            value="jazz"
          />
          <label
            className="game__check"
            htmlFor={2}
          >
            Отметить
          </label>
        </div>
      </div>
      <div
        className="track"
      >
        <button
          className="track__button track__button--play"
          disabled={true}
          onClick={[Function]}
          type="button"
        />
        <div
          className="track__status"
        >
          <audio />
        </div>
        <div
          className="game__answer"
        >
          <input
            className="game__input visually-hidden"
            id={3}
            name="answer-3"
            type="checkbox"
            value="rock"
          />
          <label
            className="game__check"
            htmlFor={3}
          >
            Отметить
          </label>
        </div>
      </div>
      <button
        className="game__submit button"
        type="submit"
      >
        Ответить
      </button>
    </form>
  </section>
</section>
`;
